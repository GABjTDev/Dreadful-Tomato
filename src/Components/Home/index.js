import React from 'react'
import MOVIES from '../../assets/movies.png'
import SERIES from '../../assets/series.png'
import { useNavigate } from 'react-router-dom';
import Header from '../Header';

const Home = () => {

  let navigate = useNavigate();

  const handleContain = (type) => {
    navigate(`/catalog/${type}`);
  }

  return (
    <>
      <Header />
      <main className='main'>
        <div className='main--contain movies' data-testid='movies' onClick={() => handleContain('movie')}>
          <img src={MOVIES} alt="cover movies" />
          <div className='main--contain-text'>
            <h2>Movies</h2>
          </div>
        </div>
        <div className='main--contain series' data-testid='series' onClick={() => handleContain('series')}>
          <img src={SERIES} alt="cover series" />
          <div className='main--contain-text'>
            <h2>Series</h2>
          </div>
        </div>
      </main>
    </>
  )
}

export default Home