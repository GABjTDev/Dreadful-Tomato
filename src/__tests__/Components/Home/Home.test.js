import React from 'react';
import Home from '../../../Components/Home';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

describe('test home page', () => {

    test('Snapshot home', () => {
        const history = createMemoryHistory();
        history.push('/');
        const wrapper = render(
            <BrowserRouter history={history}>
                <Home />
            </BrowserRouter>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Obtain img series and movies, format png', () => {
        const history = createMemoryHistory();
        history.push('/');
        const wrapper = render(
            <BrowserRouter history={history}>
                <Home />
            </BrowserRouter>
        );

        const srcImgSeries = wrapper.getByAltText('cover series');
        expect(srcImgSeries).toHaveAttribute('src', 'series.png');

        const srcImgMovies = wrapper.getByAltText('cover movies');
        expect(srcImgMovies).toHaveAttribute('src', 'movies.png');
    });

});