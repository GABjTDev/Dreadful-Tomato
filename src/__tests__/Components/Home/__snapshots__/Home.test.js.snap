// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home page Snapshot home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="container header"
      >
        <div
          class="header--logo"
        >
          <a
            href="/"
          >
            <img
              alt="Logo de Dreadfull Tomato"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="header--login"
        >
          <ul>
            <li
              class="login"
            >
              Login
              <img
                alt="icon de login"
                src="icon-login.png"
              />
            </li>
            <li
              class="trialFree"
            >
              Star you free trial
            </li>
          </ul>
        </div>
        <div
          class="header--menu"
        >
          <img
            alt="icon de menu"
            src="icon-menu.png"
          />
        </div>
      </header>
      <main
        class="main"
      >
        <div
          class="main--contain movies"
          data-testid="movies"
        >
          <img
            alt="cover movies"
            src="movies.png"
          />
          <div
            class="main--contain-text"
          >
            <h2>
              Movies
            </h2>
          </div>
        </div>
        <div
          class="main--contain series"
          data-testid="series"
        >
          <img
            alt="cover series"
            src="series.png"
          />
          <div
            class="main--contain-text"
          >
            <h2>
              Series
            </h2>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="container header"
    >
      <div
        class="header--logo"
      >
        <a
          href="/"
        >
          <img
            alt="Logo de Dreadfull Tomato"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="header--login"
      >
        <ul>
          <li
            class="login"
          >
            Login
            <img
              alt="icon de login"
              src="icon-login.png"
            />
          </li>
          <li
            class="trialFree"
          >
            Star you free trial
          </li>
        </ul>
      </div>
      <div
        class="header--menu"
      >
        <img
          alt="icon de menu"
          src="icon-menu.png"
        />
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="main--contain movies"
        data-testid="movies"
      >
        <img
          alt="cover movies"
          src="movies.png"
        />
        <div
          class="main--contain-text"
        >
          <h2>
            Movies
          </h2>
        </div>
      </div>
      <div
        class="main--contain series"
        data-testid="series"
      >
        <img
          alt="cover series"
          src="series.png"
        />
        <div
          class="main--contain-text"
        >
          <h2>
            Series
          </h2>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
